domain/consensus/model/externalapi/transactionEqualClone_test.go

```diff
@@ -0,0 +1,177 @@
+package externalapi
+
+import (
+	"reflect"
+	"testing"
+)
+
+type TransactionToCompare struct {
+	tx             *DomainTransaction
+	expectedResult bool
+}
+
+type TestDomainTransactionStruct struct {
+	baseTx                 *DomainTransaction
+	transactionToCompareTo []*TransactionToCompare
+}
+
+func InitTestBaseTransaction() *DomainTransaction {
+
+	testTx := &DomainTransaction{
+		Version:      1,
+		Inputs:       nil,
+		Outputs:      []*DomainTransactionOutput{},
+		LockTime:     1,
+		SubnetworkID: DomainSubnetworkID{0x01},
+		Gas:          1,
+		PayloadHash: DomainHash{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00},
+		Payload: []byte{0x01},
+		Fee:     0,
+		Mass:    1,
+		ID: &DomainTransactionID{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02},
+	}
+	return testTx
+}
+
+func InitTestTransactionToCompare() []*TransactionToCompare {
+
+	testTx := []*TransactionToCompare{{
```

Comment:
You need to check that every field that is different from the base will cause `.Equal` to return `false`. You can check it by running `TestDomainTransaction_Equal` and check that the test coverage of `func (tx *DomainTransaction) Equal(other *DomainTransaction) bool` is 100%. This comment also applies to any other `_Equal` test